src/routes/tools/degree-days/_data.js

```diff
@@ -0,0 +1,239 @@
+const getObservedSeries = ({ climvarId }) => {
+  return OBSERVED.map((d) => {
+    const slugs = [`${climvarId}_day_${d.id}`];
+    return { ...d, slugs, mark: "line", visible: true };
+  });
+};
+
+// For each model, there are usually 2 raster series in the API,
+// the modeled historical (1950-2005) and modeled projections (2006-2099/2021)
+const getModelSeries = ({ climvarId, scenarioId, modelIds, indicatorId }) => {
+  return PRIORITY_10_MODELS.filter((d) => modelIds.includes(d.id)).map((d) => {
+    const slugs = [
+      // `${climvarId}_day_${d.id}_historical`,
+      `${climvarId}_day_${d.id}_${scenarioId}`,
+    ];
+    return { ...d, slugs, mark: "line", visible: true, indicatorId };
+  });
+};
+
+// The ensemble has to be assembled from the max and min of 10/all models
+const getEnsembleSeries = ({ scenarioId, indicatorId }) => {
+  return ENSEMBLES.filter((d) => d.id === `${scenarioId}_range`).map((d) => {
+    const slugs = [
+      `${indicatorId}_year_ens32avg_historical`,
+      `${indicatorId}_year_ens32avg_${scenarioId}`,
+    ];
+    return { ...d, slugs, mark: "area", visible: true };
+  });
+};
+
+// determines next URL path segment following the slug
+const urlPathAfterSlug = (indicatorId, isEnsemble) => {
+  if (["cdd", "hdd"].includes(indicatorId) && !isEnsemble) return indicatorId;
+  return "events";
+};
+
+const filterReponseByMonths = (response, monthIds) =>
+  response.filter(({ date }) => monthIds.has(date.getMonth()));
+
+const aggregateMonthlyData = (response) =>
+  rollups(
+    response,
+    (arr) => +format(".0f")(mean(arr, (d) => d.value)),
+    (dd) => dd.date.getFullYear()
+  ).map(([year, value]) => ({ date: new Date(Date.UTC(year)), value }));
+
+/**
+ * Fetches data from the events endpoint in Cal-Adapt API
+ * Input parameters:
+ * slug - name of raster series in API
+ * params - object with props for geometry, stat, units, etc.
+ * method - default is GET, POST used for user uploaded boundaries
+ * @param {object}
+ * @return {array}
+ */
+const fetchEvents = async ({
+  slug,
+  params,
+  method = "GET",
+  indicatorId,
+  monthIds,
+  isEnsemble,
+}) => {
+  const url = `${apiEndpoint}/series/${slug}/${urlPathAfterSlug(
+    indicatorId,
+    isEnsemble
+  )}/`;
+  const [response, error] = await handleXHR(fetchData(url, params, method));
+  if (error) {
+    throw new Error(error.message);
+  }
+  if (monthIds && monthIds.size) {
+    return aggregateMonthlyData(
+      filterReponseByMonths(transformResponse(response), monthIds)
+    );
+  }
+  return transformResponse(response);
+};
+
+/**
+ * Fetches data for a single observed/model/ensemble series
+ * Each series can have 1 or more slugs
+ * @param {object} series - a distinct timeseries obj
+ * @param {object} params - props for for geometry, stat, units, etc.
+ * @param {string} method - default is GET, POST for uploaded boundaries
+ * @return {array}
+ */
+const fetchSeries = async ({
```

Comment:
This function might be a good spot to backfill any missing data with 0. The year ranges for the `livneh (1950-2005)` and `models (2006-2099)` are standard and can be added to the list of constants

Reply:
Ah right, I had forgotten about that part! Will make sure to backfill for any missing dates. 

We may also want to consider a `0` value and `no data` value separately and not synonymous, as technically zero is a value while missing data is just missing data.